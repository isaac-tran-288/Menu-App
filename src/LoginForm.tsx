import React from 'react';
import {
    Dialog,
    DialogActions,
    DialogContent,
    DialogTitle,
    TextField,
    Button,
} from '@mui/material';

interface FormProps {
    open: boolean,
    handleClose: () => void
}
export default function LoginForm(props: FormProps) {
    return (
        <Dialog open={props.open} onClose={props.handleClose}>
            <DialogTitle>Login</DialogTitle>
            <DialogContent>
                <TextField
                    autoFocus
                    margin="dense"
                    id="username"
                    label="Username"
                    type="email"
                    fullWidth
                    variant="standard"
                />
                <TextField
                    autoFocus
                    margin="dense"
                    id="password"
                    label="Password"
                    type="password"
                    fullWidth
                    variant="standard"
                />
            </DialogContent>
            <DialogActions>
                <Button onClick={props.handleClose}>Cancel</Button>
                <Button onClick={props.handleClose}>Subscribe</Button>
            </DialogActions>
        </Dialog>
    );
}