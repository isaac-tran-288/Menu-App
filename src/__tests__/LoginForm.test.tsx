import { shallow } from 'enzyme';
import LoginForm from '../LoginForm';
import TextField from '@mui/material/TextField';

beforeAll(() => {
    require("whatwg-fetch");
});

describe('Test Login Form', () => {
    const handleCloseMock = jest.fn();


    const wrapper = shallow(<LoginForm open={true} handleClose={handleCloseMock} />);


    test('should render 2 <TextField /> elements', () => {
        expect(wrapper.find(TextField)).toHaveLength(2);
        expect(wrapper.find(TextField).at(0).props().label).toEqual("Username");
        expect(wrapper.find(TextField).at(1).props().label).toEqual("Password");
    });


    test('should call handle close function when clicking cancel', () => {
        wrapper.find("#cancel").simulate('click');
        expect(handleCloseMock).toHaveBeenCalled();
    });

    test('should render 1 submit button', () => {
        expect(wrapper.find("#submit")).toHaveLength(1);
    });


});