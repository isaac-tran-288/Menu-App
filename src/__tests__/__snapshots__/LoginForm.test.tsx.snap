// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Login Form matches snapshot 1`] = `
<ForwardRef(Dialog)
  onClose={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  open={true}
>
  <ForwardRef(DialogTitle)>
    Login
  </ForwardRef(DialogTitle)>
  <ForwardRef(DialogContent)>
    <ForwardRef(TextField)
      autoFocus={true}
      fullWidth={true}
      id="username"
      label="Username"
      margin="dense"
      name="username"
      onChange={[Function]}
      type="email"
      value=""
      variant="standard"
    />
    <ForwardRef(TextField)
      fullWidth={true}
      id="password"
      label="Password"
      margin="dense"
      name="password"
      onChange={[Function]}
      type="password"
      value=""
      variant="standard"
    />
  </ForwardRef(DialogContent)>
  <ForwardRef(DialogActions)>
    <ForwardRef(Button)
      id="cancel"
      onClick={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
    >
      Cancel
    </ForwardRef(Button)>
    <ForwardRef(Button)
      id="submit"
      onClick={[Function]}
    >
      Submit
    </ForwardRef(Button)>
  </ForwardRef(DialogActions)>
</ForwardRef(Dialog)>
`;
